slm_porosity/heat_transfer.f90

```diff
@@ -335,58 +342,52 @@ FUNCTION user_rhs (u_integrated, p)
     IF (IMEXswitch .GE. 0) THEN
        CALL c_diff_fast(enthalpy_fusion(u_integrated(:,ie)), du, du_dummy, j_lev, ng, meth, 10, ne, 1, ne)
        T = temperature(u_integrated(:,ie))
-       du(ie,:,:) = du(ie,:,:) * SPREAD(conductivity(T) / capacity(T) * porosity_term(), 2, dim)
+       phi = liquid_fraction(u_integrated(:,ie))
+       du(ie,:,:) = du(ie,:,:) * SPREAD(conductivity(T, phi) / capacity(T, phi) * porosity_term(), 2, dim)
        CALL c_diff_fast(du(ie,:,:), d2u, d2u_dummy, j_lev, ng, meth, 10, dim, 1, dim)
        DO i = 1, dim
           user_rhs(shift+1:shift+ng) = user_rhs(shift+1:shift+ng) + d2u(i,:,i)
        END DO
     END IF
-    !PRINT *, 'user_rhs', user_rhs
-
   END FUNCTION user_rhs
 
   FUNCTION user_Drhs (pert_u, u_prev, meth)
     IMPLICIT NONE
     INTEGER, INTENT(IN) :: meth
     REAL (pr), DIMENSION (ng,ne) :: pert_u, u_prev
     REAL (pr), DIMENSION (n) :: user_Drhs
-    REAL (pr), DIMENSION (ng) :: T
+    REAL (pr), DIMENSION (ng) :: T, phi, Dphi, DT, k, c_p, Dk, Dc_p
     INTEGER :: ie, shift, i
-    REAL (pr) :: first_der
     REAL (pr), DIMENSION (ng,2*ne) :: for_du
     REAL (pr), DIMENSION (2*ne,ng,dim) :: du, du_dummy ! der(pert_u) in du(1:ne,:,:) and der(u) in du(ne+1:2*ne,:,:)
     REAL (pr), DIMENSION (dim,ng,dim) :: d2u, d2u_dummy
     REAL (pr), DIMENSION (ng,dim) :: part1, part2
 
     ie = n_var_enthalpy
     shift = ng*(ie-1)
-    first_der = conductivity_der - 2*capacity_der
     IF (IMEXswitch.LE.0) THEN
        user_Drhs(shift+1:shift+ng) = 0.0_pr
     END IF
     IF (IMEXswitch .GE. 0) THEN
+       T = temperature(pert_u(:,ie))
```

Comment:
Again, you read inattentively my comments and do not examine thoroughly your code after commit. Who will do this work instead of me? Time to grow up and bear responsibility for each letter that you write on this Earth!